jobs/dap-collector-ppa-dev/dap_collector_ppa_dev/main.py

```diff
@@ -154,6 +165,23 @@ async def collect_once(task, timestamp, duration, hpke_private_key, auth_token):
     return res
 
 
+def parse_histogram(histogram_str):
+    count_strs = histogram_str.split(",")
+    return [
+        correct_wraparound(int(count_str))
+        for count_str in count_strs
+    ]
+
+
+def correct_wraparound(num):
+    cutoff = 2 ** (FIELD_SIZE - 1)
+
+    if num > cutoff:
+        return num - FIELD_PRIME
+
+    return num
```

Comment:
We could potentially use a field from the tasks.json to decide how we performed the wraparound, looks like we have a "metric_type" field there with "vector" as the value. If we did that, then we could log an error result in the BQ table if we started using a different metric type that we didn't know how to correct.

Also don't feel strongly for this, since I haven't heard any peep about using a different metric type & as is will set us up for running the test campaign

Reply:
This is a good call, but let's do it in a future PR. The rest of the code already assumes we're gonna get a vector (and even assumed we would get a vector of size 5, which this PR now changes).

IMO it would be better to do this holistically (i.e. have none of the task details hard-coded) than to do it for a single method.